backend/main/web-api/main-web-api.yaml

```diff
@@ -0,0 +1,441 @@
+openapi: "3.0.3"
+info:
+  title: Nishiki Main Web API
+  description: Nishiki main
+  version: 1.0.0
+servers:
+  - url: www
+tags:
+  - name: user
+    description: user of Nishiki system
+paths:
+  /users/:
+    post:
+      description: Create a new user
+      tags:
+        - user
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - name
+                - emailAddress
+              properties:
+                name:
+                  description: username. this name is not provided, the name will be set as a default value.
+                  type: string
+                  nullable: true
+                emailAddress:
+                  type: string
+      responses:
+        "201":
+          description: user created
+  /users/{userId}:
+    get:
+      description: Get a user
+      tags:
+        - user
+      parameters:
+        - name: userId
+          in: path
+          description: ID of user
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: get user information
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/User"
+    put:
+      description: Update a user data
+      tags:
+        - user
+      parameters:
+        - name: userId
+          in: path
+          description: ID of user
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                name:
+                  description: user's name
+                  type: string
+      responses:
+        "202":
+          description: update user's information
+        "401":
+          description: try to change other user's information
+    delete:
+      description: delete a user
+      tags:
+        - user
```

Comment:
I found a Semantic Error here. It needs to define the declared path parameter "userId" in this delete operation.
```yml
      parameters:
        - name: userId
          in: path
          description: ID of user
          required: true
          schema:
            type: string
```